streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredWindowStore.java

```diff
@@ -1,180 +0,0 @@
-/**
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.kafka.streams.state.internals;
-
-import org.apache.kafka.common.metrics.Sensor;
-import org.apache.kafka.common.utils.Time;
-import org.apache.kafka.streams.KeyValue;
-import org.apache.kafka.streams.processor.ProcessorContext;
-import org.apache.kafka.streams.processor.StateStore;
-import org.apache.kafka.streams.processor.internals.StreamsMetricsImpl;
-import org.apache.kafka.streams.state.WindowStore;
-import org.apache.kafka.streams.state.WindowStoreIterator;
-
-public class MeteredWindowStore<K, V> implements WindowStore<K, V> {
```

Comment:
I was a bit pondering on this function: for window and session store RocksDB impl, its wrapping hierarchy is a bit different with the key-value store, as it is

```
caching -> window -> metered-segmented -> changelog-segmented -> raw-segmented (rocksdb)
```

Where key-value store hierarchy is 

```
caching -> metered -> changelog -> rocksdb
```

If users provide a customized window store implementation, I think it would be similar to key-value as

```
caching -> metered -> changelogged -> user-provided window-store
```

I think case we still need to provide the `MeteredWindowStore` and `ChangeLoggingWindowStore` wrappers. WDYT?